kornia/sensor/camera/camera.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any, Union
+
+from kornia.core import Tensor, stack, zeros_like
+from kornia.geometry.vector import Vector2, Vector3
+from kornia.image import ImageSize
+from kornia.sensor.camera.distortion import AffineTransform, BrownConradyTransform, KannalaBrandtK3Transform
+from kornia.sensor.camera.projection import OrthographicProjection, Z1Projection
+
+
+class CameraModelType(Enum):
+    PINHOLE = 0
+    BROWN_CONRADY = 1
+    KANNALA_BRANDT_K3 = 2
+    ORTHOGRAPHIC = 3
+
+
+def get_model_from_type(model_type: CameraModelType, image_size: ImageSize, params: Tensor) -> CameraModelVariants:
+    if model_type == CameraModelType.PINHOLE:
+        if params.shape[-1] != 4:
+            raise ValueError("params must be of shape B, 4 for PINHOLE Camera")
+        return PinholeModel(image_size, params)
+    elif model_type == CameraModelType.BROWN_CONRADY:
+        if params.shape[-1] != 12:
+            raise ValueError("params must be of shape B, 12 for BROWN_CONRADY Camera")
+        return BrownConradyModel(image_size, params)
+    elif model_type == CameraModelType.KANNALA_BRANDT_K3:
+        if params.shape[-1] != 8:
+            raise ValueError("params must be of shape B, 8 for KANNALA_BRANDT_K3 Camera")
+        return KannalaBrandtK3(image_size, params)
+    elif model_type == CameraModelType.ORTHOGRAPHIC:
+        if params.shape[-1] != 4:
+            raise ValueError("params must be of shape B, 4 for ORTHOGRAPHIC Camera")
+        return Orthographic(image_size, params)
+    else:
+        raise ValueError("Invalid Camera Model Type")
+
+
+# TO DO change to |
+CameraDistortionType = Union[AffineTransform, BrownConradyTransform, KannalaBrandtK3Transform]
+CameraProjectionType = Union[Z1Projection, OrthographicProjection]
+
+
+class CameraModelT:
+    def __init__(
+        self, distortion: CameraDistortionType, projection: CameraProjectionType, image_size: ImageSize, params: Tensor
+    ) -> None:
+        self.distortion = distortion
+        self.projection = projection
+        self._image_size = image_size
+        self._height = image_size.height
+        self._width = image_size.width
+        self._params = params
+        self._fx = params[..., 0]
```

Comment:
if we will keep params, is valid unpack here instead of doing it into each property?